test/hex_repo_tests.erl

```diff
@@ -0,0 +1,33 @@
+-module(hex_repo_tests).
+-include_lib("eunit/include/eunit.hrl").
+-define(fixture(Path), element(2, file:read_file("test/fixtures/" ++ Path))).
+-define(CLIENT, #{adapter => hex_http_test, user_agent_string => "(test)"}).
+-define(REPO, #{uri => "https://repo.test", public_key => ?fixture("test_pub.pem")}).
+% -define(CLIENT, #{adapter => hex_http_httpc, user_agent_string => "(httpc)"}).
+% -define(REPO, #{uri => "https://repo.hex.pm", public_key => ?fixture("hexpm_pub.pem")}).
+
+get_names_test() ->
+    {ok, #{packages := Packages}} = hex_repo:get_names(?CLIENT, ?REPO),
```

Comment:
OK, I started writing README and realized that forcing users to do:

```erlang
{ok, PublicKey} = file:read_file("hexpm_pub.pem"),
Client = #{adapter => hex_http_httpc, user_agent_string => "(httpc)"},
Repo = #{uri => "https://repo.test", public_key => PublicKey},
hex_repo.get_names(Client, Repo).
```

is not great.

I'm gonna change it so that there's `get_names([{client, Client}, {repo, Repo}])` and `get_names()` defaults to httpc adapter and repo.hex.pm (with public key included in hex_repo.erl). let me know if there's a better way to do it!